### PR TITLE
Refs [TASK][28373] Create Notification fragment UI

### PR DESCRIPTION
**Pull request by quockhanhng**
## Related Tickets
- [#28373](https://edu-redmine.sun-asterisk.vn/issues/28373)

## WHAT
- Create Notification fragment UI

## HOW
- I create notification fragment and its item layout

## WHY
- To display user's notification

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Reviewer1 (KhanhNQ) | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/thooi/ui/notification/messages/MessagesFragment.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import android.view.ViewGroup
+import com.sunasterisk.thooi.R
+class MessagesFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_messages, container, false)
+    }
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/thooi/ui/notification/messages/MessagesFragment.kt]:
```suggestion
    ): View? = inflater.inflate(R.layout.fragment_messages, container, false)
```

File: app/src/main/java/com/sunasterisk/thooi/ui/notification/notifications/NotificationsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class NotificationsFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_notifications, container, false)
+    }
+}
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/thooi/ui/notification/notifications/NotificationsFragment.kt]: có thể viết extension function r dùng 
`fun ViewGroup.inflate(@LayoutRes layoutResource: Int) = LayoutInflater.from(context).inflate(layoutResource, this, false)`

File: app/src/main/res/layout/fragment_messages.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.notification.notifications.NotificationsFragment">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/messagesRecyclerView"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_messages.xml]: recycyclerMessages

File: app/src/main/res/layout/fragment_notification.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.notification.NotificationFragment">
+
+    <include
+        android:id="@+id/toolbar"
+        layout="@layout/tool_bar_layout" />
+
+    <com.google.android.material.button.MaterialButtonToggleGroup
+        android:id="@+id/toggle_button_group"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_notification.xml]: sai quy tắc đặt tên

File: app/src/main/res/layout/fragment_notification.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        android:layout_width="match_parent"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_notification.xml]: dùng 0dp hoặc math_constraint

File: app/src/main/res/layout/fragment_notification.xml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:minWidth="@dimen/dp_128"
+            android:text="@string/label_message"
+            android:textAllCaps="false"
+            app:cornerRadius="@dimen/dp_8" />
+    </com.google.android.material.button.MaterialButtonToggleGroup>
+
+    <androidx.viewpager2.widget.ViewPager2
+        android:id="@+id/notificationViewPager"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_notification.xml]:
```suggestion
        android:id="@+id/viewPagerNotification"
```

**Review by tuannm-1276 — APPROVED** [review]

File: app/src/main/res/layout/fragment_notification.xml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        android:layout_width="match_parent"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_notification.xml]: như cmt phía trên

File: app/src/main/res/layout/fragment_notifications.xml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.notification.notifications.NotificationsFragment">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/notificationsRecyclerView"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_notifications.xml]: như cmt phía trên nữa

File: app/src/main/java/com/sunasterisk/thooi/ui/notification/notifications/NotificationsFragment.kt (hunk 2 above)
Comment (quockhanhng) [reply app/src/main/java/com/sunasterisk/thooi/ui/notification/notifications/NotificationsFragment.kt]: mấy cái fragment này là em mới gen ra thôi ạ, còn sau dùng sẽ implement BaseFragment

File: app/src/main/java/com/sunasterisk/thooi/ui/notification/ViewPagerAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+            MessagesFragment()
+        }
+    }
+
+    companion object {
+        const val NUMBER_OF_TAB = 2
+        const val NOTIFICATION_TAB_POSITION = 0
+        const val MESSAGE_TAB_POSITION = 1
+    }
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/thooi/ui/notification/ViewPagerAdapter.kt]: Truyền list fragment vào thay vì fix cứng adapter

**Pull request merged**